src/features/courses/course-controller.ts

```diff
@@ -3009,6 +3023,13 @@ class CourseController {
             [`$questions.grades.${StudentGrade.rawAttributes.userId.field}$`]: userId,
         }).omitBy(_.isNil).value() as sequelize.WhereOptions;
 
+        if (userRole === Role.STUDENT) {
+            // TODO: Fix this.
```

Comment:
The typing on this was a little weird. Above, in the other call, it was explicitly set to WhereOptions but it was inferred as WhereAttributesHash so I explicitly tagged this as one.